lua/nvim_lsp/dartls.lua

```diff
@@ -5,12 +5,12 @@ local server_name = "dartls"
 local bin_name = "dart"
 
 local find_dart_sdk_root_path = function()
-  if vim.fn["executable"]("dart") == 1 then
-    return vim.fn["resolve"](vim.fn["exepath"]("dart"))
-  elseif vim.fn["executable"]("flutter") == 1 then
+  if vim.fn["executable"]("flutter") == 1 then
     local flutter_path = vim.fn["resolve"](vim.fn["exepath"]("flutter"))
     local flutter_bin = vim.fn["fnamemodify"](flutter_path, ":h")
     return flutter_bin.."/cache/dart-sdk/bin/dart"
+  elseif vim.fn["executable"]("dart") == 1 then
+    return vim.fn["resolve"](vim.fn["exepath"]("dart"))
```

Comment:
for all of these cases, is there a reason we aren't using `vim.fn.foo()` instead of `vim.fn["foo"]()` ?

Reply:
I just changed the order it was like this previously. Don't know if there is any reason for that.